server/app/services/applicant/ApplicantData.java

```diff
@@ -66,84 +76,185 @@ public void setPreferredLocale(Locale locale) {
+    // specifically check the old path exists but not the new one.
+    Path newPath = WellKnownPaths.APPLICANT_DOB;
+    Path oldPath = WellKnownPaths.APPLICANT_DOB_DEPRECATED;
+    Path dobPath = hasPath(oldPath) && !hasPath(newPath) ? oldPath : newPath;
+    putDate(dobPath, dateOfBirth);
+  }
+
+  /**
+   * Parses a name string to extract the first, middle, and last names, if they exists, and sets
+   * those fields.
+   *
+   * @param displayName A string that contains the applicant's name, with first, middle, and last
+   *     separated by spaces. May provide only first name or only first last.
+   * @param overwrite Overwrite any existing data.
+   */
+  public void setUserName(String displayName, boolean overwrite) {
     String firstName;
-    String lastName = null;
-    String middleName = null;
+    Optional<String> lastName = Optional.empty();
+    Optional<String> middleName = Optional.empty();
     List<String> listSplit = Splitter.on(' ').splitToList(displayName);
     switch (listSplit.size()) {
       case 2:
         firstName = listSplit.get(0);
-        lastName = listSplit.get(1);
+        lastName = Optional.of(listSplit.get(1));
         break;
       case 3:
         firstName = listSplit.get(0);
-        middleName = listSplit.get(1);
-        lastName = listSplit.get(2);
+        middleName = Optional.of(listSplit.get(1));
+        lastName = Optional.of(listSplit.get(2));
         break;
       case 1:
         // fallthrough
       default:
         // Too many names - put them all in first name.
         firstName = displayName;
     }
-    setUserName(firstName, middleName, lastName);
+    setUserName(firstName, middleName, lastName, overwrite);
   }
 
+  public void setUserName(String displayName) {
+    setUserName(displayName, true);
+  }
+
+  // By default, overwrite name fields if data exists in them
+  public void setUserName(
+      String firstName, Optional<String> middleName, Optional<String> lastName) {
+    setUserName(firstName, middleName, lastName, true);
+  }
+
+  /**
+   * Sets the first, middle, and last name fields.
+   *
+   * @param firstName First name of applicant
+   * @param middleName Middle name of applicant
+   * @param lastName Last name of applicant
+   * @param overwrite When false, if first name already exists, do not update fields and return
+   *     unchanged.
+   */
   public void setUserName(
-      String firstName, @Nullable String middleName, @Nullable String lastName) {
-    if (!hasPath(WellKnownPaths.APPLICANT_FIRST_NAME)) {
-      putString(WellKnownPaths.APPLICANT_FIRST_NAME, firstName);
+      String firstName, Optional<String> middleName, Optional<String> lastName, boolean overwrite) {
+    Path firstPath = WellKnownPaths.APPLICANT_FIRST_NAME;
+    Path middlePath = WellKnownPaths.APPLICANT_MIDDLE_NAME;
+    Path lastPath = WellKnownPaths.APPLICANT_LAST_NAME;
+    if (!overwrite
+        && (applicant.getFirstName().isPresent()
```

Comment:
suggestion (non-blocking): this logic is hard to parse. maybe save it in a var called `haveFirstName` first and use that?